File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int curr = 0;
        int window = Integer.MAX_VALUE;

        int start = 0;
        int end = 0;

        for(end=0; end<nums.length; end++){
            curr += nums[end];
            while(curr >= target){
                window = Math.min(window, end-start+1);

                curr -= nums[start];
                start++;
            }
        }
        return window == Integer.MAX_VALUE ? 0 : window;
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        min_len = float('inf')
        i = 0
        curr = 0

        for j in range(n):
            curr += nums[j]

            while curr >= target:
                min_len = min(min_len, j-i+1)

                curr -= nums[i]
                i += 1

        return min_len if min_len != float('inf') else 0